lms/djangoapps/courseware/views/views.py

```diff
@@ -161,15 +161,30 @@ def courses(request):
 
     programs_list = get_programs_with_type(include_hidden=False)
 
-    return render_to_response(
+    cpa_cookie = {}
+    if not request.COOKIES.get('cpa_cookie'):
+        for var in request.GET:
+            cpa_cookie[var] = request.GET[var]
```

Comment:
Interesting, .get() is really slower https://stackoverflow.com/questions/58650208/why-is-key-in-dict-faster-than-dict-getkey-in-python3